packages/components/timepicker/src/Timepicker.test.tsx

```diff
@@ -58,56 +53,54 @@ describe('TimePicker', () => {
     ])(
       'falls back to value if input is invalid',
       (input, value, expectedDisplayValue) => {
-        const [renderResult] = build({ value });
-        fireChangeEvent(getByTestId(renderResult, 'time'), input);
-        fireBlurEvent(getByTestId(renderResult, 'time'));
-        expect(getByTestId(renderResult, 'time').value).toBe(
-          expectedDisplayValue,
+        render(
+          <Timepicker
+            date={format(addYears(new Date(), 1), 'yyyy/MM/dd')}
+            onChange={jest.fn}
+            value={value}
+          />,
         );
+
+        const timepicker = screen.getByTestId('time');
+        userEvent.type(timepicker, input);
+        timepicker.blur();
+
+        expect(timepicker).toHaveValue(expectedDisplayValue);
       },
     );
 
-    it('increases the time by half an hour on arrow up', () => {
-      const [renderResult, props] = build({ value: '12:00' });
-      fireEvent.focus(getByTestId(renderResult, 'time'));
-      fireEvent.keyUp(getByTestId(renderResult, 'time'), {
-        key: 'up arrow',
-        keyCode: 38,
-      });
-      // @ts-ignore
-      expect(props.onChange).toHaveBeenCalledWith('23:30');
+    it('decreases the time by half an hour on arrow up', () => {
+      const onChange = jest.fn();
+      render(
+        <Timepicker
+          date={format(addYears(new Date(), 1), 'yyyy/MM/dd')}
+          onChange={onChange}
+          value="12:00"
+        />,
+      );
+
+      const timepicker = screen.getByTestId('time');
+      userEvent.type(timepicker, '{arrowup}');
+
+      expect(onChange).toHaveBeenCalledWith('23:30');
+      expect(timepicker).toHaveValue('11:30 PM');
     });
 
     it('increases the time by half an hour on arrow down', () => {
-      const [renderResult, props] = build({ value: '12:00' });
-      fireEvent.focus(getByTestId(renderResult, 'time'));
-      fireEvent.keyUp(getByTestId(renderResult, 'time'), {
-        key: 'down arrow',
-        keyCode: 40,
-      });
-      // @ts-ignore
-      expect(props.onChange).toHaveBeenCalledWith('00:30');
+      const onChange = jest.fn();
+      render(
+        <Timepicker
+          date={format(addYears(new Date(), 1), 'yyyy/MM/dd')}
+          onChange={onChange}
+          value="12:00"
+        />,
+      );
+
+      const timepicker = screen.getByTestId('time');
+      userEvent.type(timepicker, '{arrowdown}');
```

Comment:
I think this works, but wondering if you tried `userEvent.keyboard(...)` and if it worked?

Reply:
Yeah I did it, it works, but `userEvent.type()` is easier here since you don't have to first use `userEvent.click()` to focus the input 😄